Doc/library/tkinter.rst

```diff
@@ -11,100 +11,267 @@
-the Tk GUI toolkit.  Both Tk and :mod:`tkinter` are available on most Unix
-platforms, as well as on Windows systems.  (Tk itself is not part of Python; it
-is maintained at ActiveState.)
+the Tcl/Tk GUI toolkit. Tcl/Tk and :mod:`tkinter` are available on most Unix
+platforms (including macOS), as well as on Windows systems.
 
 Running ``python -m tkinter`` from the command line should open a window
 demonstrating a simple Tk interface, letting you know that :mod:`tkinter` is
 properly installed on your system, and also showing what version of Tcl/Tk is
 installed, so you can read the Tcl/Tk documentation specific to that version.
 
-.. seealso::
-
-   Tkinter documentation:
-
-   `Python Tkinter Resources <https://wiki.python.org/moin/TkInter>`_
-      The Python Tkinter Topic Guide provides a great deal of information on using Tk
-      from Python and links to other sources of information on Tk.
-
-   `TKDocs <http://www.tkdocs.com/>`_
-      Extensive tutorial plus friendlier widget pages for some of the widgets.
-
-   `Tkinter reference: a GUI for Python <https://infohost.nmt.edu/tcc/help/pubs/tkinter/web/index.html>`_
-      On-line reference material.
-
-   `Tkinter docs from effbot <http://effbot.org/tkinterbook/>`_
-      Online reference for tkinter supported by effbot.org.
-
-   `Programming Python <http://learning-python.com/about-pp4e.html>`_
-      Book by Mark Lutz, has excellent coverage of Tkinter.
-
-   `Modern Tkinter for Busy Python Developers <https://www.amazon.com/Modern-Tkinter-Python-Developers-ebook/dp/B0071QDNLO/>`_
-      Book by Mark Rozerman about building attractive and modern graphical user interfaces with Python and Tkinter.
-
-   `Python and Tkinter Programming <https://www.manning.com/books/python-and-tkinter-programming>`_
-      Book by John Grayson (ISBN 1-884777-81-3).
-
-   Tcl/Tk documentation:
+Tkinter supports a range of Tcl/Tk versions, built either with or
+without thread support. The official Python binary release bundles Tcl/Tk 8.6
+threaded. See the source code for the :mode:`_tkinter` module
+for more information about supported versions.
+
+Tkinter is not a thin wrapper, but adds a fair amount of its own logic to
+make the experience more pythonic. This documentation will concentrate on these
+additions and changes, and refer to the official Tcl/Tk documentation for
+descriptions of underlying features.
+
+
+Architecture
+------------
+
+Tkinter provides a unified interface to several separate and distinct libraries.
+
+Tcl
+   Like Python, Tcl is a dynamic interpreted programming language. It is most
+   commonly used by C applications to embed a scripting language, or as an
+   interface to the Tk toolkit. The Tcl library has a C interface to
+   create and manage one or more instances of a Tcl interpreter, run Tcl
+   commands and scripts in those instances, and add custom commands
+   implemented in either Tcl or C. Tcl provides for an event queue, used
+   for I/O, timers, and by modules.
```

Comment:
This one I disagree with. The examples seem concrete to me, and by saying that it's available for multiple different things including for modules gives us info about how Tk hooks into it. This was actually following your lead a bit in that most people tend to associate event queues and event loops with a GUI toolkit, so the fact that it's part of the underlying language is unexpected. It actually used to be in Tk rather than Tcl, but that was a while back.

Regarding the omission, is it that you're reading it as the event queue is separate from the interpreter? Would saying "The Tcl interpreter maintains an event queue..." be clearer? The separate instances thing I don't quite get.

Reply:
The problem is, this is not useful information for a Tkinter user, and is confusing:
"What? How in the world can an event queue be used for I/O (it's not like the OS is listening to it, right?), or timers? (looks for an explanation further on, and doesn't find it)".
What "used by modules" means is a mystery to me. The general meaning of "module" is "some separate code chunk". Then it's basically saying "it can be used by code", which is a remark worthy of Captain Obvious.

> Regarding the omission

No, I read it as "doesn't say about the relationship at all".

> The separate instances thing I don't quite get.

In other languages, queues are just another data structures that can be created and used as one sees fit. This "Tcl queue" is not like them, it's special, and the text fails to address this.

Reply:
... https://stackoverflow.com/a/185789/648265 %)

Reply:
https://github.com/roseman/cpython/pull/5